### PR TITLE
KIALI-2454 Validate service and deployment for ports

### PR DESCRIPTION
**Pull request by burmanm**
** Describe the change **

Validates that the Service has the same port definition as available from a Deployment that matches the Service.

Also, adds validation to the Service for proper port naming. Backwards compatible without UI changes.

** Issue reference **

KIALI-2454

### CODE REVIEW

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#853](https://paste.fedoraproject.org/paste/RySEcakKDrjNU1ybK4ZAMQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1667]](https://paste.fedoraproject.org/paste/Sy7TwZh57cXUY40sO4UoTw)

**Comment by xeviknal** [comment]
Haven't finished the review yet. But couple of things:
* error messages are missing in here: https://github.com/kiali/kiali/blob/master/models/istio_validation.go#L81
* swagger documentation not updated
* question: should be a good idea to move your kiali.io PR forward? There were problems with paths and so on.

**Comment by burmanm** [comment]
Those error messages are not missing? You're comparing to master ``istio_validations.go``. Here's the PR's version, there's the ``service.deployment.port.mismatch`` (and port one was already in the master version also).

https://github.com/kiali/kiali/blob/7cd8ba22b1cc01c1e18b934e0c316e1a08c94505/models/istio_validation.go#L182

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#888](https://paste.fedoraproject.org/paste/5ETElbBQrq7pUCbsNBssUw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1690]](https://paste.fedoraproject.org/paste/gD14XqESNu9Bd~KXycclZg)

File: business/checkers/service_checker.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package checkers
+
+import (
+	"github.com/kiali/kiali/business/checkers/services"
+	"github.com/kiali/kiali/models"
+	v1beta1 "k8s.io/api/apps/v1beta1"
```
Comment (xeviknal) [inline business/checkers/service_checker.go]: Oh, nitpicking. I just realized that those imports don't follow the structure. Same thing in the following checker file.

**Comment by xeviknal** [comment]
Related to the UI, do you mind either adding a JIRA for the front-end part (not to forget) or perhaps implement a first iteration with those changes? We have been following the approach of implementing a first version and then adding the changes we get from UI team.

**Comment by burmanm** [comment]
Well, the service list is implemented in the PR but yea I can implement the sign next to the ports also.

**Comment by burmanm** [comment]
This is the UI-PR: https://github.com/kiali/kiali-ui/pull/1111

**Comment by burmanm** [comment]
Rebased

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#911](https://paste.fedoraproject.org/paste/BtE6Qtc4HI-OICw86hX3NQ)
 :red_circle: PRT Failed, Please Contact QE

**Comment by hhovsepy** [comment]
Error while building Kiali:
# github.com/kiali/kiali/kubernetes/kubetest
kubernetes/kubetest/mock.go:378:96: undefined: v1beta1
make: *** [build] Error 2

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#914](https://paste.fedoraproject.org/paste/m40uJiGCDB-C86DtV79Txg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1719]](https://paste.fedoraproject.org/paste/nFuHgxDi1aOBAkv28z4tBA)

**Comment by hhovsepy** [comment]
Build worked, but I confirm the issue described in https://github.com/kiali/kiali-ui/pull/1111#pullrequestreview-226599608

**Pull request merged**